modules/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/ignite/IgniteNodeAutoConfiguration.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.ignite;
+
+import org.apache.ignite.Ignite;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Ignite node auto configuration.
+ * Spring will use this class to autoconfigure Ignite node on application startup.
+ *
+ * Initialization of the node will be the following:
+ * <ul>
+ *   <li>If {@link IgniteConfiguration} bean exists in {@link BeanFactory} it will be used for the node start.</li>
+ *   <li>If {@link IgniteConfiguration} bean doesn't exists following rules are applied:
+ *     <ol>
+ *       <li>If bean of the type {@link IgniteConfigurer} exists in {@link BeanFactory}
+ *         it will be used to customize empty {@link IgniteConfiguration} instance.
+ *         If you want to set custom SPI instances or similar hardcoded values.
+ *         You should do it in custom {@link IgniteConfigurer}.</li>
+ *       <li>Application properties will override config values. Prefix for properties names is "ignite".</li>
+ *     </ol>
+ *   </li>
+ * </ul>
+ *
+ * Example of application property:
+ * <pre>
+ *   ignite.igniteInstanceName=my-instance
+ * </pre>
+ *
+ * @see IgniteConfigurer
+ * @see IgniteConfiguration
+ * @see Ignite
+ */
+@Configuration
+@EnableConfigurationProperties
+public class IgniteNodeAutoConfiguration {
```

Comment:
Shall we call the class name as IgniteClusterNodeAutoConfiguration since the interface in Ignite is ClusterNode ?

https://github.com/apache/ignite/blob/master/modules/core/src/main/java/org/apache/ignite/cluster/ClusterNode.java

Reply:
I think we should just remove `Node` from the class name and go with the `IgniteAutoConfiguration`. Is it OK?